src/ProjectNeon/Assets/Features/Effects/InterceptAttack.cs

```diff
@@ -0,0 +1,29 @@
+﻿
+using UnityEngine;
+
+class InterceptAttack : Effect
+{
+    private Member _performer;
+    private Target _effectTarget;
+
+    void Effect.Apply(Member source, Target target)
+    {
+        _performer = source;
+        _effectTarget = target;
+        BattleEvent.Subscribe<AttackToPerform>(attackToPerform => Execute(attackToPerform.Attack), this);
+    }
+
+    void Execute(Attack attack)
+    {
+        _effectTarget.Members.ForEach(
+            target => {
+                if (target.Equals(attack.Target.Members[0]))
+                {
+                    attack.Effect = new NoEffect();
+                    new Attack(attack.Damage).Apply(attack.Attacker, _performer);
```

Comment:
Same problem with damage being passed to the `Attack`.

The amount will be calculated based on the Attacked and the Target stats.